ScreenToGif/Capture/BaseCapture.cs

```diff
@@ -147,23 +155,51 @@ public virtual Task<int> ManualCaptureAsync(FrameInfo frame, bool showCursor = f
             return showCursor ? CaptureWithCursorAsync(frame) : CaptureAsync(frame);
         }
 
+        /// <summary>
+        /// Safely release all the managed things.
+        /// </summary>
         public virtual async Task Stop()
         {
             if (!WasStarted)
                 return;
 
-            //Stop the consumer thread.
-            BlockingCollection.CompleteAdding();
-
-            await Task.WhenAll(_task);
+            using (_task)
```

Comment:
Task doesn't need to be disposed.
Disposable resources used by task are only created if explicitely `Wait`ed, etc.
Here the `await _task` below is sufficient.
[Do I need to dispose of Tasks?](https://devblogs.microsoft.com/pfxteam/do-i-need-to-dispose-of-tasks/) is still valid.

Also I don't see any need of the finalizer at all. If an instance of this type isn't diposed properly, it's up to the BlockingCollection to clean up it's resources.
Without the finalizer we put less pressure on the GC's finalizer-queue (which is generally a good thing).

BTW: in https://github.com/NickeManarin/ScreenToGif/blob/d865e5c76e4fee3b03b3e26049da5d9cba85338a/ScreenToGif/Capture/BaseCapture.cs#L106-L107 I'd replace the `while (true)`-loop with a construct that uses `BlockingCollection.TryTake`.

Reply:
@gfoidl: In my previous version, I've changed this to BlockingCollection.TryTake, but there's a bug——If I change to this, when pressing "Stop", we cannot enter the Editor.

Reply:
I don't think we need finalizer because we don't have anything unmanaged...However due to the author's original meaning to release the managed source, I still keep it here for his view to decide whether we need it really or not.

Reply:
Thanks for your "Do I need to dispose of Tasks?", it's very useful!

Reply:
> @gfoidl: In my previous version, I've changed this to BlockingCollection.TryTake, but there's a bug——If I change to this, when pressing "Stop", we cannot enter the Editor.

The issue with the implementation is that the thread spawns and gets to the TryTake, but the collection is empty, so the thread simply ends.

Reply:
May look ugly, but it feels like this results in better performance:

```cs
while(BlockingCollection.TryTake(out var frame) || !BlockingCollection.IsCompleted)
    if (frame != null)
        Save(frame);
```

Reply:
We should switch to Channels, which are async all the way.
But I fear they're not supported on all target-frameworks.

Instead of the `while`-loop, switch to a `foreach`-loop over `GetConsumingEnumerable`? A CancellationToken could be passed in there too. Then all that handling is delegated to the framework.

Reply:
@NickeManarin  & @gfoidl: In fact, this below is enough:

```c#
while (!BlockingCollection.IsCompleted)  // This will make your BlockingCollection both empty and marked as "Completed", // so when your BlockingCollection is both empty and marked as completed, no need to loop again.
{
  FrameInfo fr = BlockingCollection.Take(); 
}
```

But it seems there's no obvious to take care of such issues, try...catch... is enough for us to solve problems. 

Ref: https://docs.microsoft.com/en-us/dotnet/standard/collections/thread-safe/blockingcollection-overview